dotnet/src/Connectors/VectorData.Abstractions/ConnectorSupport/VectorStoreRecordVectorPropertyModel.cs

```diff
@@ -54,6 +60,147 @@ public int Dimensions
     /// <seealso cref="Microsoft.Extensions.VectorData.DistanceFunction"/>
     public string? DistanceFunction { get; set; }
 
+    /// <summary>
+    /// If <see cref="EmbeddingGenerator"/> is set, contatins the type representing the embedding stored in the database.
+    /// Otherwise, this property is identical to <see cref="Type"/>.
+    /// </summary>
+    public Type EmbeddingType { get; set; } = null!;
+
+    /// <summary>
+    /// The embedding generator to use for this property.
+    /// </summary>
+    public IEmbeddingGenerator? EmbeddingGenerator { get; set; }
+
+    /// <summary>
+    /// Checks whether the <see cref="EmbeddingGenerator"/> configured on this property supports the given embedding type.
+    /// The implementation on this non-generic <see cref="VectorStoreRecordVectorPropertyModel"/> checks for <see cref="string"/>
+    /// and <see cref="DataContent"/> as input types for <see cref="EmbeddingGenerator"/>.
+    /// </summary>
+    public virtual bool TrySetupEmbeddingGeneration<TEmbedding, TUnwrappedEmbedding>(IEmbeddingGenerator embeddingGenerator, Type? embeddingType)
```

Comment:
See previous comments on how this all works. Note that this method gets overridden by the generic subclass of VectorStoreRecordVectorPropertyModel - that's how we can support arbitrary user input types.

Note that if we transition to using Embedding as the representation for raw embeddings (we currently use e.g. `ReadOnlyMemory<float>`), TUnwrappedEmbedding can go away.